src/core/solutions/dixonp.cpp

```diff
@@ -0,0 +1,54 @@
+#include <math.h>
+#include <core/solutions/dixonp.h>
+
+namespace dnn_opt
+{
+namespace core
+{
+namespace solutions
+{
+
+dixonp* dixonp::make(generator *generator, unsigned int size)
+{
+  auto* result = new dixonp(generator, size);
+
+  result->init();
+
+  return result;
+}
+
+float dixonp::calculate_fitness()
+{
+  float result = 0;
+  float* params = get_params();
+  float sum = 0;
+
+  solution::calculate_fitness();
+  
+  int length = size();
+  float binom = pow(params[0] + 1.0f, 2.0f);
+
+  for(int i = 1; i < length; i++)
+  {
+    sum += i * pow((2 * pow(params[i], 2.0f) - params[i - 1]), 2.0f);
```

Comment:
I think there is an extra parenthesis in the first parameter of the pow function

Reply:
Maybe but i use the parenthesis as a guide to divide the function base to the exp

Reply:
Anyway is a matter of style, but in this case I think it will meake the sintax more complex and difficult to read. I guess it depends of you, do as you think it will be better.